golang/cosmos/x/lien/lien.go

```diff
@@ -0,0 +1,140 @@
+// Package lien is a cosmos-sdk module that implements liens.
+// Liens are an encumbrance that prevents the transfer of tokens
+// out of an account, much like the unvested tokens in a vesting
+// account.  See spec/ for full details.
+package lien
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Agoric/agoric-sdk/golang/cosmos/vm"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// portHandler implements a vm.PortHandler.
+type portHandler struct {
+	keeper Keeper
+}
+
+// portMessage is a struct that any lien bridge message can unmarshal into.
+type portMessage struct {
+	Type    string `json:"type"`
+	Address string `json:"address"`
+	Denom   string `json:"denom"`
+	Amount  string `json:"amount"`
+}
+
+// msgAccountState marshals into the AccountState message for the lien bridge.
+type msgAccountState struct {
+	CurrentTime string `json:"currentTime"`
```

Comment:
The `CurrentTime` should be an `int64`, whatever its interpretation is.

Reply:
Done - assuming UNIX epoch.